### PR TITLE
Mask ORAS

### PR DESCRIPTION
**Pull request by philippgri11**

### CODE REVIEW

File: src/fuxi.py [hunk 1]
```diff
@@ -39,27 +35,31 @@ def forward(self, x):
...
                 outputs.append(out.detach().cpu())
-            # TODO hier muss beachtet werden, dass es für ORAS auch NaN Werte auf den Kontinenten gibt
-            # ungefähr so
-            # mask = torch.isnan( target)
-            # loss = mse( target[mask], pred[mask])
-            loss += torch.nn.functional.l1_loss(out * lat_weights, timeseries[:, step + 2, :, :, :] * lat_weights)
+
+            # Maske für die aktuelle Zeitschritt anwenden
+            step_mask = mask[:, step + 2, :, :, :]
+            loss += torch.nn.functional.l1_loss(out * lat_weights * step_mask,
```
Comment (max-ksoll) [inline src/fuxi.py]: Hier bin ich mir gerade nicht ganz sicher, ob wir nicht l1_loss(out*lat, ts*lat, red=sum)*mask / mask.sum()
machen müssten

File: src/fuxi.py [hunk 2]
```diff
@@ -121,12 +121,13 @@ def __init__(self, in_channels, out_channels):
...
         self.layers = torch.nn.ModuleList([self.conv1, self.residual_block])
+        self.conv_adjust = nn.Conv2d(out_channels * 2, out_channels, kernel_size=1)
 
     def forward(self, x):
         out = self.conv1(x)
         residual = self.residual_block(out)
-        # TODO durch concat ersetzen
-        out = out + residual
+        out = torch.cat((out, residual), dim=1)
+        out = self.conv_adjust(out)
```
Comment (max-ksoll) [inline src/fuxi.py:130]: Hier hätte ich gedacht, dass wir dem ersten Transformer eben genau das torch.cat ergebniss geben.
Ist wahrscheinlich am Ende aber eh egal

File: src/fuxi.py (hunk 1 above)
Comment (philippgri11) [reply src/fuxi.py]: Also ich hatte es so gedacht:
Angenommen, wir haben die folgenden Daten:

out: Vorhersagen des Modells, z.B. tensor([[[[1.0, 2.0], [3.0, 4.0]]]])
timeseries: Zielwerte, z.B. tensor([[[[1.0, NaN], [3.0, 4.5]]]])
lat_weights: Gewichtungen, z.B. tensor([[[1.0, 1.0], [1.0, 1.0]]])
	
step_mask = tensor([[[[True, False], [True, True]]]])
out_masked = out * lat_weights * step_mask
# out_masked = tensor([[[[1.0, 0.0], [3.0, 4.0]]]])

target_masked = timeseries * lat_weights * step_mask
# target_masked = tensor([[[[1.0, 0.0], [3.0, 4.5]]]])

l1_loss = torch.nn.functional.l1_loss(out_masked, target_masked, reduction='sum')
# l1_loss = abs(1.0 - 1.0) + abs(0.0 - 0.0) + abs(3.0 - 3.0) + abs(4.0 - 4.5)
# l1_loss = 0.0 + 0.0 + 0.0 + 0.5
# l1_loss = 0.5

normalized_loss = l1_loss / step_mask.sum()
# normalized_loss = 0.5 / 3
# normalized_loss = 0.1667

Comment (max-ksoll) [reply src/fuxi.py]: Was hälst du davon es so zu schreiben
sum(abs(ts-out) x lat_weights x mask) / mask.sum()
Da würden wir doch weniger multiplikationen machen
